src/verifier.rs

```diff
@@ -5,20 +5,65 @@ use std::sync::Mutex;
 
 use bitcoin::{BlockHash, TxOut};
 use bitcoin::blockdata::block::Block;
-use bitcoin::hashes::Hash;
 use lightning::log_error;
 use lightning::routing::gossip::{NetworkGraph, P2PGossipSync};
 use lightning::routing::utxo::{UtxoFuture, UtxoLookup, UtxoResult, UtxoLookupError};
 use lightning::util::logger::Logger;
 use lightning_block_sync::{BlockData, BlockSource};
+use lightning_block_sync::gossip::UtxoSource;
 use lightning_block_sync::http::BinaryResponse;
 use lightning_block_sync::rest::RestClient;
+use lightning_block_sync::rpc::RpcClient;
 
 use crate::config;
 use crate::types::GossipPeerManager;
 
+enum AnyChainSource {
+	Rest(RestClient),
+	Rpc(RpcClient),
+}
```

Comment:
Could we use a type parameter instead of introducing an enum?

Reply:
Yes, but it bleeds into the entire application and felt unnecessary when RGS is a binary and not really a lib.  But I can go back and do this if you think it's better once we resolve the GossipVerifier issues.